reactor-core/src/main/java/reactor/core/scheduler/InstantPeriodicWorkerTask.java

```diff
@@ -70,7 +70,7 @@ public Void call() {
 		try {
 			try {
 				task.run();
-				setRest(executor.submit(this));
+				setRest(executor.submit((Callable<Void>) this));
```

Comment:
should this cast to `Runnable` instead? (see discussion in the `HookableScheduledExecutorService`)

`Runnable` is the only type that is guaranteed to be maintained by the `Function<Runnable, Runnable>`... `Callable` is used internally mostly for short-circuiting subscription IIRC, but I don't think they've been actively used for tasks scheduling purposes so far.

Having this new hook kind of enforces us to limit our use of `ExecutorService` to `Runnable`-based methods... cc @smaldini

Reply:
I did it to align with the rest of tasks. AFAIR the problem is that `ScheduledThreadPoolExecutor` wraps Runnable into Callable when submitted:
https://github.com/frohoff/jdk8u-jdk/blob/da0da73ab82ed714dc5be94acd2f0d00fbdfe2e9/src/share/classes/java/util/concurrent/FutureTask.java#L152